pkg/appctx/container.go

```diff
@@ -24,6 +27,18 @@ func (e ErrNoItemFound) Error() string {
 	return fmt.Sprintf("no item with key %v found", e.Key)
 }
 
+type KeyLock struct {
```

Comment:
I'm wondering if it would be possible to delete the mutex on unlocking to free up the memory again?